### PR TITLE
Run full tests only on Wednesday scheduled jobs

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Address comment https://github.com/GenericMappingTools/pygmt/issues/1686#issuecomment-1061357814.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
It should work, but I can't test it.

**Comment by weiji14** [comment]
I was thinking of running these full tests on a less regular basis, maybe weekly or monthly instead?

**Comment by seisman** [comment]
> I was thinking of running these full tests on a less regular basis, maybe weekly or monthly instead?

I agree. Changing the following lines to weekly?

https://github.com/GenericMappingTools/pygmt/blob/813142bbc2ef84b49cff91f20dcb99999ce067d6/.github/workflows/ci_tests.yaml#L21

**Comment by weiji14** [comment]
> > I was thinking of running these full tests on a less regular basis, maybe weekly or monthly instead?
> 
> I agree. Changing the following lines to weekly?
> 
> https://github.com/GenericMappingTools/pygmt/blob/813142bbc2ef84b49cff91f20dcb99999ce067d6/.github/workflows/ci_tests.yaml#L21

But this means the daily unit tests will become weekly. 

Or is there a way to just run the doctests, and not the unit tests? That way we could put it in the docs workflow as a weekly/monthly scheduled run (in addition to the current docs build).

**Comment by seisman** [comment]
> > > I was thinking of running these full tests on a less regular basis, maybe weekly or monthly instead?
> > 
> > 
> > I agree. Changing the following lines to weekly?
> > https://github.com/GenericMappingTools/pygmt/blob/813142bbc2ef84b49cff91f20dcb99999ce067d6/.github/workflows/ci_tests.yaml#L21
> 
> But this means the daily unit tests will become weekly.

We have PRs merging into the main branch every few days, so the unit tests are also run every few days, making the daily unit tests less necessary. So a weekly job is still acceptable to me.

> Or is there a way to just run the doctests, and not the unit tests? That way we could put it in the docs workflow as a weekly/monthly scheduled run (in addition to the current docs build).

Sphinx has its own doctest extension, which can run doctests.
```
cd doc/
sphinx-build -b doctest . _build
```
The above command works, but it reports many "errors" that are not reported by pytest.

File: .github/workflows/ci_tests.yaml
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:143]: Not sure if GitHub Actions uses `&` or `and`, but try something like this to run full tests on Wednesday only. Xref https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule
```suggestion
        if: github.event_name == 'schedule' & github.event.schedule == '0 0 * * 3'
```

Comment (seisman) [reply .github/workflows/ci_tests.yaml]: > Not sure if GitHub Actions uses `&` or `and`,

It's `&&` (https://docs.github.com/en/actions/learn-github-actions/expressions#operators).

File: .github/workflows/ci_tests.yaml
Comment (seisman) [inline .github/workflows/ci_tests.yaml:138]: This `if` statement means the tests won't be run in daily jobs, which is not what we want. However, removing this `if` statement means on Wednesday job, most tests will be run twice by both `make test` and `make fulltest`.

Comment (weiji14) [reply .github/workflows/ci_tests.yaml:138]: What about something like
```suggestion
        if: github.event.schedule != '0 0 * * 3'
```
Though not sure if it will error for regular (non-scheduled) tests, e.g. in PRs.

File: .github/workflows/ci_tests.yaml
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:141]:
```suggestion
      # Run full tests including doctests on Wednesday
```

**Review by weiji14 — COMMENTED** [review]
Also need to update `doc/maintenance.md` to mention the new full tests schedule.

File: .github/workflows/ci_tests.yaml
Comment (seisman) [reply .github/workflows/ci_tests.yaml]: No errors in PRs.

File: .github/workflows/ci_tests.yaml
Comment (weiji14) [inline .github/workflows/ci_tests.yaml:136]:
```suggestion
      # Run regular tests
```

**Review by weiji14 — APPROVED** [review]
Nice! Just one last suggestion, and wait for second approval to merge.

**Pull request merged**
